spec/lib/mantle/message_spec.rb

```diff
@@ -17,7 +17,23 @@
 
       mantle_message.publish(message)
     end
-  end
-end
 
+    it "published message includes message_source" do
+      Mantle.configure { |config| config.whoami = 'SantaClaus' }
+      bus = double("message bus")
+      catch_up = double("catch up")
+      channel = "create:person"
+      message = { id: 1 }
+      actual_message = message.merge(message_source: 'SantaClaus')
 
+      mantle_message = Mantle::Message.new(channel)
+      mantle_message.message_bus = bus
+      mantle_message.catch_up = catch_up
+
+      expect(bus).to receive(:publish).with(channel, actual_message)
```

Comment:
set up `bus` with `allow`:

`allow(bus).to receive(:publish)`

Then run your test, and use `expect` to assert the message was received:

`expect(bus).to have_received(:publish).with(channel, actual_message)`

If I learned anything from working with Thoughtbot for 4 months, it's this!